modules/godbcbackend/sodbc.cc

```diff
@@ -0,0 +1,414 @@
+      errmsg<<state<<i<<native<<text<<"/";
+    }
+  }
+  while( ret == SQL_SUCCESS );
+  throw SSqlException( errmsg.str() );
+}
+
+class SODBCStatement: public SSqlStatement
+{
+public:
+  SODBCStatement(const string& query, bool dolog, int nparams, SQLHDBC connection)
+  {
+    SQLRETURN result;
+
+    d_query = query;
+    // d_nparams = nparams;
+    d_conn = connection;
+    d_dolog = dolog;
+    d_residx = 0;
+    d_paridx = 0;
+
+    // Allocate statement handle.
+    result = SQLAllocHandle( SQL_HANDLE_STMT, d_conn, &d_statement );
+    testResult( result, SQL_HANDLE_DBC, d_conn, "Could not allocate a statement handle." );
+
+    result = SQLPrepare(d_statement, (SQLCHAR *) query.c_str(), SQL_NTS);
+    testResult( result, SQL_HANDLE_STMT, d_statement, "Could not prepare query." );
+
+    SQLSMALLINT paramcount;
+    result = SQLNumParams(d_statement, &paramcount);
+    testResult( result, SQL_HANDLE_STMT, d_statement, "Could not get parameter count." );
+
+    if (paramcount != nparams)
+      throw SSqlException("Provided parameter count does not match statement: " + d_query);
+
+    d_parnum = nparams;
+    // cerr<<"prepared ("<<query<<")"<<endl;
+  }
+
+  typedef struct {
+    SQLPOINTER      ParameterValuePtr;
+  } ODBCParam;
+
+  vector<ODBCParam> d_req_bind;
+
+  SSqlStatement* bind(const string& name, bool value) { return bind(name, (long)value); }
+  SSqlStatement* bind(const string& name, long value) { 
+
+    // cerr<<"asked to bind long "<<value<<endl;
+    // cerr<<"d_req_bind.size()="<<d_req_bind.size()<<endl;
+    // cerr<<"d_parnum="<<d_parnum<<endl;
+
+    if(d_req_bind.size() > (d_parnum+1)) throw SSqlException("Trying to bind too many parameters.");
+
+    ODBCParam p;
+
+    p.ParameterValuePtr = new long[1];
+    *((long*)p.ParameterValuePtr) = value;
+
+    d_req_bind.push_back(p);
+
+    SQLRETURN result = SQLBindParameter(
+      d_statement,           // StatementHandle,
+      d_paridx+1,            // ParameterNumber,
+      SQL_PARAM_INPUT,       // InputOutputType,
+      SQL_C_SLONG,           // ValueType,
+      SQL_BIGINT,            // ParameterType,
+      0,                     // ColumnSize,
+      0,                     // DecimalDigits,
+      p.ParameterValuePtr,  // ParameterValuePtr,
+      0,                     // BufferLength,
+      NULL                      // StrLen_or_IndPtr
+    );
+    testResult( result, SQL_HANDLE_STMT, d_statement, "Binding parameter.");
+    d_paridx++;
+
+    return this;
+  }
+  SSqlStatement* bind(const string& name, uint32_t value) { return bind(name, (long)value); }
+  SSqlStatement* bind(const string& name, int value) { return bind(name, (long)value); }
+  SSqlStatement* bind(const string& name, unsigned long value) { return bind(name, (long)value);}
+  SSqlStatement* bind(const string& name, long long value) { return bind(name, (long)value); };
+  SSqlStatement* bind(const string& name, unsigned long long value) { return bind(name, (long)value); }
```

Comment:
Perhaps not cast everything to long, but handle them correctly? This might bite later on.